webdriver-spec.html

```diff
@@ -5777,19 +5778,25 @@
  <li><p>Let <var>body</var> and <var>arguments</var>
   be <var>script arguments</var>’ data.
 
- <li><p>Let <var>result</var> be the result of calling <a>execute a function body</a>,
-  with arguments <var>body</var> and <var>arguments</var>.
+ <li><p>Let <var>promise</var> be a new Promise.
 
- <li><p>Wait for <a>execute a function body</a> to return
-  or for <a>session script timeout</a> milliseconds to expire,
-  whichever happens sooner.
+ <li><p>Run the following substeps <a>in parallel</a>:
+   <ol>
+     <li><p>Let <var>result</var> be the result of calling <a>execute a function body</a>,
+      with arguments <var>body</var> and <var>arguments</var>.
 
- <li><p>If the previous step completed due to the <a>session script timeout</a> being reached,
-  return <a>error</a> with <a>error code</a> <a>script timeout</a>.
+     <li><p>If <var>promise</var> is still pending and
```

Comment:
This step should run in parallel with function execution, so it should it should be part of the parent list.

1. Let `promise` be a new Promise.
2.  Run the following steps in parallel:
      1. Let `result` be the result of calling execute a function body...
      2. If `result` is an error...
      3. Otherwise resolve `promise` with `result`
3. If `promise` is still pending and `session script timeout` milliseconds is reached...

I might even put the timeout in its own parallel track:


1. Let `promise` be a new Promise.
2.  Run the following steps in parallel:
      1. Let `result` be the result of calling execute a function body...
      2. If `result` is an error...
      3. Otherwise resolve `promise` with `result`
3. Run the following steps in parallel:
      1. Wait `session script timeout` milliseconds
       2. Reject `promise` with an `error` with `error code` `script timeout`